src/Domain/DomainHelpers.cpp

```diff
@@ -874,6 +874,12 @@ maps_for_rectilinear_domains(
   std::vector<std::unique_ptr<
       CoordinateMapBase<Frame::Logical, TargetFrame, VolumeDim>>>
       maps{};
+  // Tracks the value of the collapsed index of the IndexIterator ii,
```

Comment:
I found the use of the word `Tracks` confusing. Maybe start with something like "almost_collapsed_index is equal to IndexIterator.collapsed_index() except in the case where...."

Reply:
I think `almost_collapsed_index` should just be renamed `block_id` or `block_index` as that is what it corresponds to.